src/ClientData/CallstackDataTest.cpp

```diff
@@ -200,4 +200,144 @@ TEST(CallstackData, FilterCallstackEventsBasedOnMajorityStartExcludesFunctionToS
                                                                          event4, event5, event6}));
 }
 
+TEST(CallstackData, ForEachCallstackEventOfTidInTimeRangeDiscretized) {
+  CallstackData callstack_data;
+
+  constexpr uint32_t kTid = 42;
+  constexpr uint32_t kAnotherTid = 43;
+
+  constexpr uint64_t kCloneAddress = 0x10;
+
+  const uint64_t cs1_id = 12;
+  const uint64_t cs1_outer = kCloneAddress;
+  const uint64_t cs1_inner = 0x11;
```

Comment:
I don't think there's any point declaring `cs#_outer`, `cs#_inner` as you are using them once and don't care about the values of the frames in this test. Just inline them in the construction of the `CallstackInfo`s.